module/zfs/vdev_raidz.c

```diff
@@ -1845,7 +1845,12 @@ raidz_parity_verify(zio_t *zio, raidz_row_t *rr)
 
 		if (!rc->rc_tried || rc->rc_error != 0)
 			continue;
-
+		if (orig[c] == NULL) {
+			vdev_raidz_checksum_error(zio, rc, orig[c]);
+			rc->rc_error = SET_ERROR(ECKSUM);
+			ret++;
+			continue;
+		}
```

Comment:
This is a false positive, although I can see why Coverity flagged it.  We're guaranteed to hit the continue above on 1847 if `orig[c] == NULL`.  It is possible `rc->rc_error` will be modified between the two loops in this function but it will never be cleared.  We can ASSERT orig[c] isn't NULL if it would satisfy the static analysis.

Reporting a checksum error also isn't quite right, that would indicate the data is wrong, but really the data is just missing since it was unreadable.

Reply:
You are right. I missed that. I am closing this.